### DISCUSSION TITLE
When does terminal.size() return an error?

### DISCUSSION THREAD

**Discussion by bunchofcellulose** [fdehau/tui-rs#592] [Q&A]
The type signature of `CrosstermBackend::size()` is `(&self) -> io::Result<Rect>`, so when does it return an error?

**Comment by PierreAntoineGuillaume** [comment]
I don't know for sure because I don't know tui-rs much, but an exemple is :

When it runs whithout being attached to a terminal.

for instance, you run it in a subshell

```shell
var=$(cargo run --exemple demo)
```  

for more info, I guess you could read into TTYs.
